libcloud/compute/drivers/openstack.py

```diff
@@ -322,14 +322,19 @@ def detach_volume(self, volume, ex_node=None):
         # when ex_node is not provided, volume is detached from all nodes
         failed_nodes = []
         for attachment in volume.extra['attachments']:
-            if not ex_node or ex_node.id == attachment['serverId']:
+            if not ex_node or ex_node.id == attachment.get(
+                'serverId'
+            ) or attachment['server_id']:
```

Comment:
This comparison is not correct.
In case of passing an ex_node, if attachment['server_id'] is not None, the attachment is deleted using any node.
It should be something like:
```
if not ex_node or ex_node.id in [attachment.get('serverId'), attachment['server_id']]
```

Reply:
oops, good catch! I'll fix that